api/src/routes/identity/index.test.ts

```diff
@@ -10,477 +10,404 @@ import { UserIdentityMock } from '../../test/mocks/identities';
-			body: null
-		};
+		it('is not authorized to delete different user', async () => {
+			const deleteUserSpy = spyOn(UserDb, 'deleteUser');
 
-		await userRoutes.deleteUser(req, res, nextMock);
+			const req: any = {
+				user: { identityId: 'did:iota:123456789' },
+				params: { identityId: 'did:iota:2QQd1DN1ZjnXnvSAaAjk1VveBNUYDw7eE9bTTCC4RbG4' },
+				body: null
+			};
 
-		expect(deleteUserSpy).toHaveBeenCalledTimes(0);
-		expect(nextMock).toHaveBeenCalledWith(new Error('not allowed!'));
-	});
+			await userRoutes.deleteUser(req, res, nextMock);
 
-	it('should delete expected user', async () => {
-		const deleteUserSpy = spyOn(UserDb, 'deleteUser');
+			expect(deleteUserSpy).toHaveBeenCalledTimes(0);
+			expect(nextMock).toHaveBeenCalledWith(new Error('not allowed!'));
+		});
 
-		const req: any = {
-			user: { identityId: 'did:iota:2QQd1DN1ZjnXnvSAaAjk1VveBNUYDw7eE9bTTCC4RbG4' },
-			params: { identityId: 'did:iota:2QQd1DN1ZjnXnvSAaAjk1VveBNUYDw7eE9bTTCC4RbG4' },
-			body: null
-		};
+		it('should delete expected user', async () => {
+			const deleteUserSpy = spyOn(UserDb, 'deleteUser');
 
-		await userRoutes.deleteUser(req, res, nextMock);
+			const req: any = {
+				user: { identityId: 'did:iota:2QQd1DN1ZjnXnvSAaAjk1VveBNUYDw7eE9bTTCC4RbG4' },
+				params: { identityId: 'did:iota:2QQd1DN1ZjnXnvSAaAjk1VveBNUYDw7eE9bTTCC4RbG4' },
+				body: null
+			};
 
-		expect(deleteUserSpy).toHaveBeenCalledTimes(1);
-		expect(sendStatusMock).toHaveBeenCalledWith(200);
-	});
+			await userRoutes.deleteUser(req, res, nextMock);
 
-	it('should call next(err) if an error occurs when removing from db', async () => {
-		const deleteUserSpy = spyOn(UserDb, 'deleteUser').and.callFake(() => {
-			throw new Error('Test error');
+			expect(deleteUserSpy).toHaveBeenCalledTimes(1);
+			expect(sendStatusMock).toHaveBeenCalledWith(200);
 		});
-		const req: any = {
-			user: { identityId: 'did:iota:2QQd1DN1ZjnXnvSAaAjk1VveBNUYDw7eE9bTTCC4RbG4' },
-			params: { identityId: 'did:iota:2QQd1DN1ZjnXnvSAaAjk1VveBNUYDw7eE9bTTCC4RbG4' },
-			body: null
-		};
-		await userRoutes.deleteUser(req, res, nextMock);
 
-		expect(deleteUserSpy).toHaveBeenCalledTimes(1);
-		expect(sendMock).not.toHaveBeenCalled();
-		expect(nextMock).toHaveBeenCalledWith(new Error('Test error'));
+		it('should call next(err) if an error occurs when removing from db', async () => {
+			const deleteUserSpy = spyOn(UserDb, 'deleteUser').and.callFake(() => {
+				throw new Error('Test error');
+			});
+			const req: any = {
+				user: { identityId: 'did:iota:2QQd1DN1ZjnXnvSAaAjk1VveBNUYDw7eE9bTTCC4RbG4' },
+				params: { identityId: 'did:iota:2QQd1DN1ZjnXnvSAaAjk1VveBNUYDw7eE9bTTCC4RbG4' },
+				body: null
+			};
+			await userRoutes.deleteUser(req, res, nextMock);
+
+			expect(deleteUserSpy).toHaveBeenCalledTimes(1);
+			expect(sendMock).not.toHaveBeenCalled();
+			expect(nextMock).toHaveBeenCalledWith(new Error('Test error'));
+		});
```

Comment:
it's quire challenging to review a file of this size with that amount of changes :)

Reply:
I am very sorry, I guess this whole PR will be a lot of code changes which are quite some effort to review. From my point of you, you could avoid test files. Explicitely this has a bigger refactored behind.